### PR TITLE
File Manager - Add context menu to thumbnails item list and extract common toolbar / context menu code to the separate module

### PR DESCRIPTION
**Pull request by vladkovl**
Key functionalities:
- context menu supported for the thumbnails view mode of the item list
- extracted common code from the toolbar / context menu to a separate module

### CODE REVIEW

File: js/ui/file_manager/ui.file_manager.command_manager.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    }
+
+    getCommands(forToolbar, items) {
+        return this._commands
+            .filter(c => (!c.displayInToolbarOnly || forToolbar) && (!items || this.isCommandAvailable(c.name, items)));
+    }
+
+    isCommandAvailable(commandName, items) {
+        const command = this._commandMap[commandName];
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.command_manager.js]:
```suggestion
        const { noFileItemRequired, isSingleFileItemCommand } = this._commandMap[commandName];
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 2]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
+        this._editActions = {
 
-    tryCreate() {
-        const item = this._model.getCurrentFolder();
-        this._actions.onCreating();
+            create: {
+                useCurrentFolder: true,
+                affectsAllItems: true,
+                dialog: this._createFolderDialog,
+                action: (items, arg) => this._provider.createFolder(items[0], arg.name),
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                action: ([item], { name }) => this._provider.createFolder(item, name),
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 3]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
-        this._tryEditAction(
-            this._createFolderDialog,
-            result => this._provider.createFolder(item, result.name),
-            "Folder created",
-            info => `Create folder operation failed for the ${item.name} parent folder`
-        );
-    }
+            rename: {
+                dialog: this._renameItemDialog,
+                getDialogArgument: items => items[0].name,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                getDialogArgument: [{ name }] => name,
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 4]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
+                action: (items, arg) => this._provider.renameItem(items[0], arg.name),
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                action: ([item], { name }) => this._provider.renameItem(item, name),
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 5]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
+                getErrorMessage: (items, info) => `Rename operation failed for the ${items[0].name} item`,
+            },
 
-    tryDelete(fileItems) {
-        if(fileItems.length === 0) {
-            return;
-        }
+            delete: {
+                dialog: this._confirmationDialog,
+                getDialogArgument: items => items[0].name,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                getDialogArgument: [{ name }]=> name,
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 6]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
+                action: (items, arg) => this._provider.deleteItems(items),
+                getSuccessMessage: items => "Item renamed",
+                getErrorMessage: (items, info) => `Delete operation failed for the ${items[info.index].name} item`
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                getErrorMessage: (items, { index }) => `Delete operation failed for the ${items[index].name} item`
```

File: js/ui/file_manager/ui.file_manager.item_list.thumbnails.js [hunk 7]
```diff
@@ -40,6 +44,8 @@ class FileManagerThumbnailsItemList extends FileManagerItemListBase {
         this.$element().addClass(FILE_MANAGER_THUMBNAILS_ITEM_LIST_CLASS);
         this.$element().append(this._$viewPort);
 
+        const eventName = addNamespace(contextMenuEventName, FILE_MANAGER_THUMBNAILS_CONTEXT_MENU_EVENT);
+        eventsEngine.on(this.$element(), eventName, this._onContextMenu.bind(this));
         eventsEngine.on(this.$element(), "click", this._getItemSelector(), this._onClick.bind(this));
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.thumbnails.js]: Add namespace?

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 8]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
-    tryCreate() {
-        const item = this._model.getCurrentFolder();
-        this._actions.onCreating();
+            create: {
+                useCurrentFolder: true,
+                affectsAllItems: true,
+                dialog: this._createFolderDialog,
+                action: ([item], { name }) => this._provider.createFolder(item, name),
+                getSuccessMessage: items => "Folder created",
+                getErrorMessage: (items, info) => `Create folder operation failed for the ${items[0].name} parent folder`
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                getErrorMessage: ([{ name }], info) => `Create folder operation failed for the ${name} parent folder`
```

File: js/ui/file_manager/ui.file_manager.editing.js [hunk 9]
```diff
@@ -90,78 +95,94 @@ class FileManagerEditingControl extends Widget {
...
-            "Folder created",
-            info => `Create folder operation failed for the ${item.name} parent folder`
-        );
-    }
+            rename: {
+                dialog: this._renameItemDialog,
+                getDialogArgument: [{ name }] => name,
+                action: ([item], { name }) => this._provider.renameItem(item, name),
+                getSuccessMessage: items => "Items deleted",
+                getErrorMessage: (items, info) => `Rename operation failed for the ${items[0].name} item`,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.editing.js]:
```suggestion
                getErrorMessage: ([{ name }], info) => `Rename operation failed for the ${name} item`,
```

**Review by DokaRus — DISMISSED** [review]

File: js/ui/file_manager/ui.file_manager.item_list.thumbnails.js (hunk 7 above)
Comment (vladkovl) [reply js/ui/file_manager/ui.file_manager.item_list.thumbnails.js]: OK. Now the namespace has been added to the "click" event too.

**Review by DokaRus — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Pull request merged**